cmd/manager/main.go

```diff
@@ -225,3 +239,50 @@ func checkIfRequiredFilesExist(requiredFiles []string) error {
 	}
 	return nil
 }
+
+// clusterConfigChecks checks for required configuration in a cluster before starting the controller.
+// If the cluster doesn`t meet the configuration, it returns an error.
+func clusterConfigChecks(cfg *rest.Config) error {
```

Comment:
nit: You could have created a struct around clusterConfig

```suggestion
func clusterConfigChecks(cfg *rest.Config) error {
type clusterConfig struct {
oclient OpenShiftClient
operatorClient
networkType <interface>
}
```